### PR TITLE
Move IDLs into sections

### PR DESCRIPTION
**Pull request by ccameron-chromium**
In the Proposed Solution section, there exist sub-sections for each part of the API (2D Canvas, WebGL, WebGPU, HTMLCanvasElement, and ImageData). Move the IDL changes for each of those API parts into those sub-sections (this is how ImageData is already, but do the same for the other parts)

### CODE REVIEW

**Comment by ccameron-chromium** [comment]
In this patch I intentionally just move things around, and don't change the language in any of the sections. The remaining changes to the spec are
- add the ImageBitmap section that we've discussed at W3C
- add examples
- do a final pass of language (it has the fingerprints of many authors)

**Comment by ccameron-chromium** [comment]
Post-weekend ping. This change is moving text around (and intentionally not changing the text -- trying to keep the "move stuff" and "change stuff" diffs separate).

File: CanvasColorSpaceProposal.md [hunk 1]
```diff
@@ -72,43 +72,47 @@ interface CanvasColorSpace {
...
 </pre>
 
 Example:
 <pre>
 canvas.getContext('2d', { colorSpace: "display-p3"} );,
 </pre>
 
-#### The ``colorSpace`` canvas creation attribute
-
 The ``colorSpace`` attribute specifies the color space for the backing storage of the canvas.
```
Comment (kenrussell) [inline CanvasColorSpaceProposal.md:82]: At this point in the new text, the "colorspace" attribute hasn't been defined in IDL yet. Should there be a mention that it's defined in subsequent sections?

**Review by kenrussell — APPROVED** [review]
Generally looks good; one thought.

File: CanvasColorSpaceProposal.md (hunk 1 above)
Comment (ccameron-chromium) [reply CanvasColorSpaceProposal.md:82]: You're right.

This is a part that will need re-working. I had started doing bits of clean-up along with the restructuring, but the diffs became unwieldy.

Once this general re-structure in is in place (and once the ImageBitmap part is added), then I'm going to do a final pass of tightening up the verbiage and fixing these sorts of dependencies.

Comment (kenrussell) [reply CanvasColorSpaceProposal.md:82]: OK, fine by me to move forward with this and continue to do cleanups.

**Comment by ccameron-chromium** [comment]
Ping on this again -- does anyone else want to review this (this re-org is blocking the changes to the verbiage of each section).

(Thanks Ken for the review!)

**Review by jdashg — APPROVED** [review]

**Comment by jdashg** [comment]
Merged, thanks!

**Pull request merged**